src/main/java/org/truffleruby/extra/ffi/PointerNodes.java

```diff
@@ -95,6 +96,7 @@ public boolean setAutorelease(DynamicObject pointer, boolean autorelease) {
 
         @Child private AllocateObjectNode allocateObjectNode = AllocateObjectNode.create();
 
+        @TruffleBoundary
         @Specialization
         public DynamicObject malloc(DynamicObject pointerClass, long size) {
             return allocateObjectNode.allocate(pointerClass, memoryManager().newPointer(getContext().getNativePlatform().getMallocFree().malloc(size)));
```

Comment:
Can you move the boundary inside, so the allocateNode is not behind?
Allocating a DynamicObject in interpreter is pretty expensive and PE can optimize it much better.